src/mon/MonitorDBStore.h

```diff
@@ -635,8 +640,16 @@ class MonitorDBStore
   }
 
   int create_and_open(ostream &out) {
+    // record the type before open
+    string kv_type;
+    int r = read_meta("kv_backend", &kv_type);
+    if (r < 0) {
+      r = write_meta("kv_backend", g_conf->mon_keyvaluedb);
+      if (r < 0)
+	return r;
+    }
```

Comment:
can we pass the `kv_type` to `_open()`, so we don't need to read it again. and it helps to consolidate the logic handling missing or empty kv_type.